### PR TITLE
workspace/notify notification creates no vscode command as expected

### LINKED ISSUE #1367: workspace/notify notification creates no vscode command as expected
I'm trying to send an notification on JDT LS extension side by sending a notification `workspace/notify` with:

https://github.com/eclipse/eclipse.jdt.ls/blob/cf6358ba4c166fa7d4c45aba5b7b22ea08516524/org.eclipse.jdt.ls.core/src/org/eclipse/jdt/ls/core/internal/lsp/ExecuteCommandProposedClient.java#L26

On TypeScript client side, I'm trying to consume the expected notification but it doesn't work although it works with `workspace/executeClientCommand` which is a request and not a notification.

It seems that there is a bug in https://github.com/redhat-developer/vscode-java/blob/master/src/extension.ts#L367

```ts
languageClient.onRequest(SendNotificationRequest.type, (params) => {
						return commands.executeCommand(params.command, ...params.arguments);
					});
```

but it should use `onNotification` (not onRequest):

```ts
languageClient.onNotification(SendNotificationRequest.type, (params) => {
						return commands.executeCommand(params.command, ...params.arguments);
					});
```

### PR DESCRIPTION
**Pull request by angelozerr**
workspace/notify notification creates no vscode command as expected

Fixes #1367

Signed-off-by: azerr azerr@redhat.com

### CODE REVIEW

File: src/protocol.ts [hunk 1]
```diff
@@ -112,8 +112,8 @@ export namespace ExecuteClientCommandRequest {
     export const type = new RequestType<ExecuteCommandParams, any, void, void>('workspace/executeClientCommand');
 }
 
-export namespace SendNotificationRequest {
-    export const type = new RequestType<ExecuteCommandParams, any, void, void>('workspace/notify');
+export namespace SendNotificationNotification {
```
Comment (fbricon) [inline src/protocol.ts:115]: ServerNotification

**Comment by fbricon** [comment]
@testforstephen @yaohaizh @jdneo please check this is compatible with vscode-java-debug, as you guys introduced it.

File: src/protocol.ts (hunk 1 above)
Comment (angelozerr) [reply src/protocol.ts]: fixed.

**Comment by testforstephen** [comment]
See https://github.com/microsoft/vscode-java-test/issues/260 and https://github.com/microsoft/vscode-java-test/pull/309/files, Test runner is consuming this api. @jdneo  Could you verify whether this break the test runner?

**Comment by jdneo** [comment]
The usage of `sendNotification()` API was removed long time ago, This PR won't break the Java Test Runner.

**Pull request merged**